### PR TITLE
Make it possible to read logback version from catalog

### LINKED ISSUE #561: MicronautBuildCommonPlugin doesn't read logback version from micronaut-logging catalog
MicronautBuildCommonPlugin executes the following code to find logback version

`def logbackVersionProvider = versionProviderOrDefault(project, 'logback', '1.2.3')`

The versionProviderOrDefault method doesn't use the micronaut-logging catalog to find the logback version so returns the default one (1.2.3).

### PR DESCRIPTION
**Pull request by melix**
This commit adds the ability to read the default logback version from the `mnLogging` catalog instead of the `libs` catalog or Gradle properties. It also removes the default version as it is likely to be outdated in any case and raise problems.

The same should probably done for objenesis and bytebuddy but it's unclear which projects use it in practice.

Fixes #561

### CODE REVIEW

File: src/main/groovy/io/micronaut/build/MicronautBuildCommonPlugin.groovy [hunk 1]
```diff
@@ -45,7 +45,7 @@ class MicronautBuildCommonPlugin implements Plugin<Project> {
         }
         def byteBuddyVersionProvider = versionProviderOrDefault(project, 'bytebuddy', '1.12.18')
         def objenesisVersionProvider = versionProviderOrDefault(project, 'objenesis', '3.1')
-        def logbackVersionProvider = versionProviderOrDefault(project, 'logback', '1.2.3')
+        def logbackVersionProvider = versionProviderOrDefault(project, 'logback', List.of("mnLogging", "libs"), '')
```
Comment (msupic) [inline src/main/groovy/io/micronaut/build/MicronautBuildCommonPlugin.groovy:48]: Should we change the order of catalogs so we can override logback version from 'mnLogging' catalog with version from 'libs' catalog if necessary?
Also, what if a project doesn't have logback in 'libs' and doesn't import micronaut-logging, will the build fail with empty default version?

Comment (sdelamo) [reply src/main/groovy/io/micronaut/build/MicronautBuildCommonPlugin.groovy:48]: The default should be `1.4.7` not `1.2.3` in master.

Comment (melix) [reply src/main/groovy/io/micronaut/build/MicronautBuildCommonPlugin.groovy:48]: I don't think that the build plugins should provide a default version. That's why I removed it altogether. The default version should be read from properties file or the catalogs, so indeed, builds will fail if it's not declared, which is a breaking change.

As for inverting the order, I'm not sure it makes sense.

Comment (melix) [reply src/main/groovy/io/micronaut/build/MicronautBuildCommonPlugin.groovy]: I have pushed a change to prioritize `libs` over `mnLogging`. In addition, the logback dependency will only be added to `testImplementation` if the logback version is defined (either explicitly or via a catalog).

**Review by sdelamo — APPROVED** [review]

**Pull request merged**